.github/workflows/rust.yml

```diff
@@ -33,3 +35,14 @@ jobs:
         with:
           name: coverage
           path: ./coverage/
+
+  build_edition_2018:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Switch to edition 2018 (Hack)
+        run: sed -i 's/2021/2018/' Cargo.toml && head -8 Cargo.toml
+      - name: Build
+        run: cargo build --verbose
+      - name: Run tests
+        run: cargo test --verbose
```

Comment:
What's this for?

Reply:
Ensure that we're using an implementation which is compatible to both 2018 and 2021. If any user wants to stick to 2018, they could still use a few new versions.

I have little knowledge of Rust edition, not sure if this is necessary. If not, I'd be more than happy to remove this hack